elide-model-config/src/main/resources/elideTableSchema.json

```diff
@@ -1,454 +1,426 @@
-          "description": "An array of string based tags for dimensions",
-          "type": "array",
-          "uniqueItems": true,
-          "items": {
-            "type": "string"
-          },
-          "default": []
-        }
-
-      }
-    },
-    "dimension": {
-      "title": "Dimension",
-      "description": "Dimensions represent labels for measures. Dimensions are used to filter and group measures.",
-      "type": "object",
-      "allOf": [
-        { "$ref": "#/definitions/dimensionRef" },
-        {
-          "properties": {
-            "type": { "$ref": "#/definitions/columnTypes" },
-            "values": {
-              "title": "Dimension values",
-              "description": "An array of valid string values for this dimension",
-              "type": "array",
-              "uniqueItems": true,
-              "items": {
-                "type": "string"
-              },
-              "default": []
+        "measure": {
+            "title": "Measure",
+            "description": "Metric definitions are extensible objects that contain a type field and one or more additional attributes. Each type is tied to logic in Elide that generates a metric function.",
+            "type": "object",
+            "properties": {
+                "name": {
+                    "title": "Metric name",
+                    "description": "The name of the metric. This will be the same as the POJO field name.",
+                    "type": "string",
+                    "format": "elideFieldName"
+                },
+                "description": {
+                    "title": "Metric description",
+                    "description": "A long description of the metric.",
+                    "type": "string"
+                },
+                "category": {
+                    "title": "Measure group category",
+                    "description": "Category for grouping",
+                    "type": "string"
+                },
+                "hidden": {
+                    "title": "Hide/Show measure",
+                    "description": "Whether this metric is exposed via API metadata",
+                    "type": "boolean",
+                    "default": false
+                },
+                "readAccess": {
+                    "title": "Measure read access",
+                    "description": "Read permission for the metric.",
+                    "type": "string",
+                    "default": "Prefab.Role.All"
+                },
+                "definition": {
+                    "title": "Metric definition",
+                    "description": "The definition of the metric",
+                    "type": "string"
+                },
+                "queryPlanResolver": {
+                    "title": "query plan resolver class",
+                    "description": "The JVM function class which maps a metric to a query plan",
+                    "pattern": "^([a-zA-Z]+[a-zA-Z0-9_]*\\.)+class",
```

Comment:
Do we need a format for this as well?

Reply:
I was going to do in next PR, along with formats required for other schemas